### PR TITLE
chore: add codespell support and make it fix few typos

### PR DESCRIPTION
**Pull request by yarikoptic**
More about codespell: https://github.com/codespell-project/codespell .  It detects (and can be used to fix) commonly made typos.  There is a good number of them found in this codebase.

I personally introduced it to dozens if not hundreds of projects already and so far only positive feedback.

CI workflow has 'permissions' set only to 'read' so also should be safe.

### CODE REVIEW

File: src/common/chunk_copies_calculator_unittest.cc
Comment (yarikoptic) [inline src/common/chunk_copies_calculator_unittest.cc:174]: note code change -- but looked reasonable to me but I didn't look too much

File: src/master/restore.cc
Comment (Antuan-AE) [inline src/master/restore.cc:838]: This suggestion modifies the intention of string comparison since size of string to compare has been hard coded. Should not be accepted.

File: src/nfs-ganesha/main.c
Comment (Antuan-AE) [inline src/nfs-ganesha/main.c:63]: This change is modifying a field name inside a structure, so it should fail at build time.

File: tests/setup_machine.sh
Comment (Antuan-AE) [inline tests/setup_machine.sh:22]: Using some kind of dictionary to find potential typos doesn't apply here where acronym is of type `hd{a,b...z}`
```suggestion
		${script_name} setup /mnt/hda /mnt/hdb
```

File: src/nfs-ganesha/main.c
Comment (yarikoptic) [reply src/nfs-ganesha/main.c:63]: is this an external API which should not be changed/fixed without major version boost or internal one? (I found no use of that field in this codebase)

Comment (yarikoptic) [reply src/nfs-ganesha/main.c:63]: is there CI to test the build? (all PRs seems "waiting" for some good old jenkins)

File: tests/setup_machine.sh
Comment (yarikoptic) [reply tests/setup_machine.sh:22]: thanks for spotting -- missed that one. Will exclude all `/mnt/[a-z0-9]+`

**Comment by yarikoptic** [comment]
thanks for the rapid review! addressed  all, but `homogenous` (waiting for an answer), and force pushed, so some comments might get "outdated".

File: src/nfs-ganesha/main.c
Comment (uristdwarf) [reply src/nfs-ganesha/main.c:63]: > is this an external API which should not be changed/fixed without major version boost or internal one? (I found no use of that field in this codebase)

Yes, this is part of the FSAL API by Ganesha NFS.

> is there CI to test the build? (all PRs seems "waiting" for some good old jenkins)

Currently no, you'll have to build it yourself. I can also run the tests manually if needed and confirm there's no breaking changes (though I doubt there will be)

You can use these commands to build it all (including Ganesha support):
`mkdir build; cd build`
`sudo cmake -DENABLE_TESTS=1 -DENABLE_DOCS=1 -DENABLE_NFS_GANESHA=1 -DENABLE_CLIENT_LIB=ON -DENABLE_URAFT=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo ..`

And then:

`make -j[number of parallel jobs]`

**Comment by uristdwarf** [comment]
Hey!

Thanks for taking the time to make this PR. This looks actually pretty cool solution for spelling problems.

I just have two suggestions:

1. Please format either the PR title or commits individually (preferably both) according to https://www.conventionalcommits.org/en/v1.0.0/. This is required because we use it to determine the version bump number. `chore` or `style` seem good. The title should be also limited to max 70 (preferably 58)
2. Please add a more descriptive description (why it's needed, what it's doing) so it's easier to later read from `git log`.

Other than, fix the issues as other reviewers have noted, and I can run the build and tests manually once it's done to confirm it's all working.

File: src/cgi/sfs.cgi.in
Comment (yarikoptic) [reply src/cgi/sfs.cgi.in]: added pragma to ignore this line

File: src/master/restore.cc
Comment (yarikoptic) [reply src/master/restore.cc:838]: added pragma to ignore this line

File: src/nfs-ganesha/main.c
Comment (yarikoptic) [reply src/nfs-ganesha/main.c:63]: added pragma to ignore this line

Automated CI build testing is nearly the first thing I set it up for any new project - wouldn't trust myself or anyone else otherwise these days ;)  If I were to start using saunafs, I would have also added it as a filesystem to test git-annex like we do already for NFS: https://github.com/datalad/git-annex/blob/master/.github/workflows/build-ubuntu.yaml#L281

**Comment by yarikoptic** [comment]
- minimally adjust commit messages
- rebased
- signed off
- added one more pragma to ignore that `homogenous` typo
- force-pushed

File: src/nfs-ganesha/main.c
Comment (uristdwarf) [reply src/nfs-ganesha/main.c]: > Automated CI build testing is nearly the first thing I set it up for any new project - wouldn't trust myself or anyone else otherwise these days ;) If I were to start using saunafs, I would have also added it as a filesystem to test git-annex like we do already for NFS: https://github.com/datalad/git-annex/blob/master/.github/workflows/build-ubuntu.yaml#L281

I mean we have it, but we have two problems.
1. The CI is sometimes a bit broken. So we have a rule if the tests pass locally, then it can be merged without CI approval.
2. Currently we don't allow running it and viewing outside the organization, so it's hard for contributors to see what's wrong.

Comment (yarikoptic) [reply src/nfs-ganesha/main.c]: If you point me to your setup for the other CI, I should be quickly replicate as a github action -- should then be safe (unless needs secrets) to run it for public PRs etc.

**Comment by yarikoptic** [comment]
I've changed base branch for the PR -- commits were already off the `dev` so no changes were needed there.

File: src/nfs-ganesha/main.c
Comment (uristdwarf) [reply src/nfs-ganesha/main.c]: I'll run it manually on the Jenkins CI, no worries.

> If you point me to your setup for the other CI, I should be quickly replicate as a github action -- should then be safe (unless needs secrets) to run it for public PRs etc.

This is actually something we are looking into at the moment. I made some internal tool to help me run the tests as quickly as possible for quick iteration. This is not published yet nor do I think it's ready for CI use, but yesterday I talked with @ralcolea and he's also working on something similar as well.

File: src/nfs-ganesha/main.c
Comment (ralcolea) [inline src/nfs-ganesha/main.c:63]: Thank you very much @yarikoptic for your contribution.
In this case, `homogenous` is the name of an attribute in `struct fsal_staticfsinfo_t`, which is a Ganesha definition. So, even if it there is a typo in the name, we should use it like it is to avoid compilation errors.

Being said that, I think adding the rule to ignore the typo is the correct solution :+1:

**Comment by yarikoptic** [comment]
I found a browser tab with this PR open... is there anything else to be done to finalize it?

**Comment by uristdwarf** [comment]
> I found a browser tab with this PR open... is there anything else to be done to finalize it?

We are still fixing CI, so right now every other PR is on hold as well

**Review by uristdwarf — DISMISSED** [review]

**Review by ralcolea — DISMISSED** [review]
Great job @yarikoptic :clap: :clap: :clap: 
Thank you very much for your contribution :+1:

File: src/nfs-ganesha/main.c
Comment (ralcolea) [reply src/nfs-ganesha/main.c]: I think this conversation is solved. I will close it.

File: tests/setup_machine.sh
Comment (ralcolea) [reply tests/setup_machine.sh]: This suggestion was addressed as @Antuan-AE suggested, so it is solved.

Comment (ralcolea) [reply tests/setup_machine.sh]: This suggestion was addressed correctly.

**Comment by yarikoptic** [comment]
- abandoned last commit (run of `codespell -w`)
- rebased on dev
- reran `codespell -w`
- force-pushed

**Review by ralcolea — APPROVED** [review]
Good job @yarikoptic

**Review by uristdwarf — APPROVED** [review]

**Pull request merged**
